tools/backup/templates/backup.yaml

```diff
@@ -26,6 +26,9 @@ spec:
             {{ $key }}: "{{ $value }}"
             {{- end }}
         spec:
+          {{- if .Values.serviceAccountName }}
+          serviceAccountName: {{ .Values.serviceAccountName }}
```

Comment:
I think you need to add this param `automountServiceAccountToken` as well with ServiceAccountName. This will mount the service account token.

Reply:
I didn't need this when I tested in GCP.

I think it's `true` by default and only needs to be explicitly set to opt out: https://kubernetes.io/docs/tasks/configure-pod-container/configure-service-account/#use-the-default-service-account-to-access-the-api-server

Reply:
Service account functionality works the same in AWS and GCP.
Yes, by default `automountServiceAccountToken` is true but in our case, we are using terraform to create SA and the default value for `automountServiceAccountToken` is false. If you enable this feature while creating SA then it will mount an SA token for each Pod you create in your namespace even though the Pod does not require AWS credentials.

Reply:
Ok fair enough, let me see what I can do